### PR TITLE
chore(rln-relay-v2): add tests for serde

### PR DESCRIPTION
**Pull request by rymnc**
# Description

This PR includes tests for the serde changes in rln-v2, including `toLeaf` and `proofGen/proofVerify`

# Changes

- [x] test for `proofGen`
- [x] kat for `toLeaf`

## How to test

1.
 `nim c --out:build/rln_v2_serde -r -d:chronicles_log_level=INFO --verbosity:0 --hints:off -d:git_version="v0.24.0-rc.0-35-g50308e" -d:release --passL:librln_v0.4.1.a --passL:-lm -d:rln_v2 tests/waku_rln_relay/rln_v2/test_rln_relay_v2_serde.nim` (provided you ran `RLN_V2=true make -j16 update` before)
OR
`RLN_V2=true make -j16 test`

### CODE REVIEW

**Review by alrevuelta — APPROVED** [review]

File: tests/waku_rln_relay/rln_v2/test_rln_relay_v2_serde.nim [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      userMessageLimit: 100
+  )
+
+suite "RLN Relay v2: serde":
+  test "toLeaf: converts a rateCommitment to a valid leaf":
+    # this test vector is from zerokit
+    let rateCommitment = defaultRateCommitment()
+
+    let leafRes = toLeaf(rateCommitment)
+    assert leafRes.isOk, $leafRes.error
```
Comment (Ivansete-status) [inline tests/waku_rln_relay/rln_v2/test_rln_relay_v2_serde.nim:50]: Nitpick. I've seen this elsewhere:)
```suggestion
    assert leafRes.isOk(), $leafRes.error
```

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks!

I added a minor comment. Aside from that, maybe it is worth adding a test example exploring "wrong" edge cases such as incorrect rateCommitment, if possible

File: waku/waku_rln_relay/conversion_utils.nim [hunk 2]
```diff
@@ -69,8 +69,8 @@ when defined(rln_v2):
     ## the serialization is done as instructed in  https://github.com/kilic/rln/blob/7ac74183f8b69b399e3bc96c1ae8ab61c026dc43/src/public.rs#L146
     ## [ id_key<32> | id_index<8> | epoch<32> | signal_len<8> | signal<var> ]
     let memIndexBytes = toBytes(uint64(memIndex), Endianness.littleEndian)
-    let userMessageLimitBytes = toBytes(uint64(userMessageLimit), Endianness.littleEndian)
-    let messageIdBytes = toBytes(uint64(messageId), Endianness.littleEndian)
+    let userMessageLimitBytes = cast[array[32, byte]](toBytes(uint64(userMessageLimit), Endianness.littleEndian))
```
Comment (arnetheduck) [inline waku/waku_rln_relay/conversion_utils.nim:72]: using `cast` is a good way to introduce undefined behavior - it should be avoided in most cases - `array[...](toBytes...)` will work iff this is a compatible conversion, otherwise you need to construct the array in some other way.

**Review by SionoiS — APPROVED** [review]
Thanks!

serde is :heart:

File: waku/waku_rln_relay/conversion_utils.nim (hunk 2 above)
Comment (rymnc) [reply waku/waku_rln_relay/conversion_utils.nim]: addressed in [4095279](https://github.com/waku-org/nwaku/pull/2421/commits/40952795c5bc7fd05047f3f1b44bab7e025b44e7)
